client/app/services/dialog-field-refresh.service.js

```diff
@@ -24,6 +24,12 @@
           }
         });
 
+        _.find(allDialogFields, function(dialogField) {
```

Comment:
Ah, well, I don't actually care about the results of the find, so this actually still works in the UI. The anonymous function that's setting the `beingRefreshed` property is all that matters, though I agree this does look a little odd and should probably be more like your second one.